tests/integration/route_test.py

```diff
@@ -436,3 +442,48 @@ def test_iface_down_with_routes_in_current(eth1_up, get_routes_func):
 
     cur_state = libnmstate.show()
     _assert_routes([], cur_state)
+
+
+@pytest.fixture(scope='function')
+def eth1_static_gateway_dns(eth1_up):
+    routes = (
+        [_get_ipv4_gateways()[0], _get_ipv6_gateways()[0]]
+        + _get_ipv4_test_routes()
+        + _get_ipv6_test_routes()
+    )
+
+    state = {
+        Interface.KEY: [ETH1_INTERFACE_STATE],
+        Route.KEY: {Route.CONFIG: routes},
+        DNS.KEY: {
+            DNS.CONFIG: {
+                DNS.SERVER: [IPV6_DNS_NAMESERVER, IPV4_DNS_NAMESERVER],
+                DNS.SEARCH: DNS_SEARCHES,
+            }
+        },
+    }
+
+    libnmstate.apply(state)
+    yield state
+    # Remove DNS config
+    libnmstate.apply(
```

Comment:
This teardown is a bit strange.
- Why are the `interfaces` mentioned? Is it a must?
- Clearing all DNS entries may wipe out other existing entries. Clearing the interface config (eth1_up should do the job) is enough, no? The routes seems to be handled the same here, otherwise I would expect you to remove them explicitly.

Reply:
* The `interfaces` is required.
* If DNS not cleared, next test will fail with `failed to find proper interface for name server` as DNS config will be merged if not defined.